go/billing/tests/test_tasks.py

```diff
@@ -425,6 +427,151 @@ def test_generate_monthly_statement_sessions_different_markups(self):
+            session_length_cost=100,
+            markup_percent=10.0)
+
+        statement = tasks.generate_monthly_statement(
+            self.account.id, *this_month())
+        [item] = get_line_items(statement).filter(
+            description='Sessions (billed per 20s)')
+
+        self.assertEqual(item.credits, get_session_length_credits(300, 10))
+        self.assertEqual(item.unit_cost, 100)
+        self.assertEqual(item.cost, 300)
+
+    def test_generate_monthly_statement_different_unit_times(self):
+        mk_transaction(
+            self.account,
+            session_unit_time=20,
+            session_length_cost=100,
+            markup_percent=10.0)
+
+        mk_transaction(
+            self.account,
+            session_unit_time=21,
+            session_length_cost=100,
+            markup_percent=10.0)
+
+        mk_transaction(
+            self.account,
+            session_unit_time=22,
+            session_length_cost=100,
+            markup_percent=10.0)
+
+        statement = tasks.generate_monthly_statement(
+            self.account.id, *this_month())
+        [item1, item2, item3] = get_line_items(statement).filter(
+            description__startswith='Sessions (billed per')
+
+        self.assertEqual(item1.credits, get_session_length_credits(100, 10))
+        self.assertEqual(item1.unit_cost, 100)
+        self.assertEqual(item1.cost, 100)
+        self.assertEqual(item1.description, 'Sessions (billed per 20s)')
+
+        self.assertEqual(item2.credits, get_session_length_credits(100, 10))
+        self.assertEqual(item2.unit_cost, 100)
+        self.assertEqual(item2.cost, 100)
+        self.assertEqual(item2.description, 'Sessions (billed per 21s)')
+
+        self.assertEqual(item3.credits, get_session_length_credits(100, 10))
+        self.assertEqual(item3.unit_cost, 100)
+        self.assertEqual(item3.cost, 100)
+        self.assertEqual(item3.description, 'Sessions (billed per 22s)')
+
+    def test_generate_monthly_statement_different_session_length_costs(self):
+        mk_transaction(
+            self.account,
+            session_unit_time=20,
+            session_length_cost=100,
+            markup_percent=10.0)
+
+        mk_transaction(
+            self.account,
+            session_unit_time=20,
+            session_length_cost=200,
+            markup_percent=10.0)
+
+        mk_transaction(
+            self.account,
+            session_unit_time=20,
+            session_length_cost=300,
+            markup_percent=10.0)
+
+        statement = tasks.generate_monthly_statement(
+            self.account.id, *this_month())
+        [item1, item2, item3] = get_line_items(statement).filter(
+            description='Sessions (billed per 20s)')
+
+        self.assertEqual(item1.credits, get_session_length_credits(100, 10))
+        self.assertEqual(item1.unit_cost, 100)
+        self.assertEqual(item1.cost, 100)
+        self.assertEqual(item1.description, 'Sessions (billed per 20s)')
```

Comment:
After discussion in RL, different ``session_length_costs`` should be grouped together into one item if ``session_unit_cost`` is the same for all of them and split out if it is different.